test_rules_scala.sh

```diff
@@ -883,3 +928,6 @@ $runner bazel build //test_expect_failure/proto_source_root/... --strict_proto_d
 $runner scala_binary_jar_is_exposed_in_build_event_protocol
 $runner scala_test_jar_is_exposed_in_build_event_protocol
 $runner scala_junit_test_jar_is_exposed_in_build_event_protocol
+$runner test_scala_import_source_jar_should_be_fetched_when_fetch_sources_is_set_to_true
+$runner test_scala_import_source_jar_should_be_fetched_when_env_bazel_jvm_fetch_sources_is_set_to_true
```

Comment:
I don't understand what's going on here. I'd imagine this test (the second one) always passed because of the previous test. Additionally I imagine the last test to always fail because the src jar is already downloaded.
What am I missing?

Reply:
in test 1 there is no env var
in test 2 the env var is set to true, which triggers re-evaluation of the rule and downloads the sources.
in test 3 we change the env var to false, which again triggers re-evaluation and removes the source jar.

All tests pass of course.